packages/builtin/array.pony

```diff
@@ -328,14 +331,40 @@ class Array[A] is Seq[A]
 
         n = n + 1
       end
+    end
+
+    n = 0
+
+    // If a concrete len is specified, we take the caller at their word
```

Comment:
This is good for performance but the cost of getting this wrong is a segfault.

Do we want to do that?